app/budget/services/budgetExcelService.js

```diff
@@ -0,0 +1,456 @@
+class BudgetExcelService {
+	constructor (ApiService) { // eslint-disable-line no-unused-vars
+		
+	}
+
+	generateDownload(viewState) {
+
+		// eslint-disable-next-line no-console
+		console.dir(viewState);
+
+		var data = [];
+		var termDescriptions = {
+			'05': 'Summer Session 1',
+			'06': 'Summer Special Session',
+			'07': 'Summer Session 2',
+			'08': 'Summer Quarter',
+			'09': 'Fall Semester',
+			'10': 'Fall Quarter',
+			'01': 'Winter Quarter',
+			'02': 'Spring Semester',
+			'03': 'Spring Quarter'
+		};
+
+// SUMMARY EXCEL REPORT
+		var summary = viewState.summary;
+
+		// Header
+		var header = [" "];
+		viewState.calculatedScheduleCosts.terms.forEach(function(term) {
+			let termDescription = termDescriptions[term];
+			header.push(termDescription);
+		});
+		header.push("Total");
+		data.push(header);
+
+		// TA Count
+		var row = [];
+		row.push("TA Count");
+		summary.terms.forEach(function(term) {
+				row.push(summary.byTerm[term].taCount);
+		});
+		row.push(summary.combinedTerms.taCount);	
+		data.push(row);
+		row = [];
+	
+		// TA Cost
+		row.push("TA Cost");
+		summary.terms.forEach(function(term) {
+				row.push(summary.byTerm[term].taCost);
+		});
+		row.push(summary.combinedTerms.taCost);	
+		data.push(row);
+		row = [];
+	
+		// Reader Count
+		row.push("Reader Count");
+		summary.terms.forEach(function(term) {
+				row.push(summary.byTerm[term].readerCount);
+		});
+		row.push(summary.combinedTerms.readerCount);	
+		data.push(row);
+		row = [];
+	
+		// Reader Cost 
+		row.push("Reader Cost ");
+		summary.terms.forEach(function(term) {
+				row.push(summary.byTerm[term].readerCost);
+		});
+		row.push(summary.combinedTerms.readerCost);	
+		data.push(row);
+		row = [];
+	
+		// Support Cost
+		row.push("Support Cost");
+		summary.terms.forEach(function(term) {
+				row.push(summary.byTerm[term].supportCosts);
+		});
+		row.push(summary.combinedTerms.supportCosts);	
+		data.push(row);
+		row = [];
+	
+		// Empty row
+		data.push([" "]);
+		
+		// Replacement Cost sub-types
+		// Support Cost
+		var instructorTypes = viewState.instructorTypes;
+		summary.combinedTerms.replacementCosts.instructorTypeIds.forEach(function(instructorTypeId) {
+				row.push(instructorTypes.list[instructorTypeId].description);
```

Comment:
Inappropriate indentation, needs a new line to separate the loop on 90 from the row.push on 89